### ISSUE TITLE
[NTI] False enum warning that rhs "is not a subtype of" lhs

### ISSUE THREAD

**Issue by mgiuffrida** [google/closure-compiler#2258]
Enums have a weird warning under NTI: "The right side in the assignment is not a subtype of the left side."

Code:

```js
/** @enum {number} */ const MyEnum = { ONE: 1, TWO: 2, };
/** @type {MyEnum} */ let num = MyEnum.ONE;
module.exports = {MyEnum: MyEnum};
```

Result:

```
% java -jar closure-compiler.jar --language_in ECMASCRIPT6_STRICT \
    --language_out ECMASCRIPT5_STRICT --warning_level VERBOSE \
    --new_type_inf --process_common_js_modules --common_js_entry_module=enums.js enums.js

enums.js:1: WARNING - The right side in the assignment is not a subtype of the left side.
Expected : module$test.MyEnum{ONE:module$test.MyEnum^, TWO:module$test.MyEnum^}
Found    : {ONE:number, TWO:number}
More details:
Incompatible types for property ONE.
Expected : module$test.MyEnum
Found    : number
/** @enum {number} */ const MyEnum = { ONE: 1, TWO: 2, }; module.exports = {MyEnum: MyEnum};
                            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^

0 error(s), 1 warning(s), 100.0% typed
```

Seems like it doesn't happen without using CommonJS modules.

**Comment by Dominator008** [comment]
Closed by https://github.com/google/closure-compiler/commit/e69590fafb3762a6a138d60b27c901f7ea124323.
